testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -1520,11 +1546,21 @@ public void testAllImmediateFailureEventsPresent()
 
     private void assertLineage(String baseQuery, Set<String> inputTables, OutputColumnMetadata... outputColumnMetadata)
             throws Exception
+    {
+        assertLineage(baseQuery, inputTables, false, outputColumnMetadata);
+    }
+
+    private void assertLineage(String baseQuery, Set<String> inputTables,boolean containsUnnest ,OutputColumnMetadata... outputColumnMetadata)
+            throws Exception
     {
         assertLineageInternal("CREATE TABLE mock.default.create_new_table AS " + baseQuery, inputTables, outputColumnMetadata);
         assertLineageInternal("CREATE VIEW mock.default.create_new_view AS " + baseQuery, inputTables, outputColumnMetadata);
         assertLineageInternal("CREATE VIEW mock.default.create_new_materialized_view AS " + baseQuery, inputTables, outputColumnMetadata);
-        assertLineageInternal("INSERT INTO mock.default.table_for_output(test_varchar, test_bigint) " + baseQuery, inputTables, outputColumnMetadata);
+        if (containsUnnest) {
+            assertLineageInternal("INSERT INTO mock.default.table_for_unnest_output(test_varchar, test_bigint, test_array_unnest) " + baseQuery, inputTables, outputColumnMetadata);
+        } else {
+            assertLineageInternal("INSERT INTO mock.default.table_for_output(test_varchar, test_bigint) " + baseQuery, inputTables, outputColumnMetadata);
+        }
```

Comment:
I don't think we need to handle separately for `containsUnnest` - the base query should have two columns one varchar and other bigint - we could have two test cases where one case is from unnest value and other is from cardinality.